### PR TITLE
chore(refactor): performance improvements

### PR DESCRIPTION
**Pull request by nattadex**
#### What this PR does / why we need it:
Moving the bridge status check api call to a `getServerSideProps()` function

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: apps/web/src/pages/index.tsx [hunk 1]
```diff
@@ -90,4 +92,20 @@ function Home() {
...
+  let isLoaded = false;
+  let isBridgeUp = true;
+  if (data) {
+    if (data?.isUp === true) {
+      isBridgeUp = true;
+    }
+    isLoaded = true;
+  }
+  return {
+    props: { isLoaded, isBridgeUp }, // will be passed to the page component as props
```
Comment (kevzzsk) [inline apps/web/src/pages/index.tsx:107]: `isBridgeUp` doesn't seem to be used anywhere. in `Home` component seems to only read `isLoaded`.

File: apps/web/src/pages/index.tsx [hunk 2]
```diff
@@ -90,4 +92,20 @@ function Home() {
   );
 }
 
+export async function getServerSideProps() {
+  const res = await fetch(`https://wallet.defichain.com/api/v0/bridge/status`);
```
Comment (kevzzsk) [inline apps/web/src/pages/index.tsx:96]: Please add the error case and http error status case for non 200 cases

File: apps/web/src/pages/index.tsx [hunk 3]
```diff
@@ -90,4 +90,18 @@ function Home() {
...
+  const data = await res.json();
+  let isBridgeUp = true;
+  if (data) {
+    if (data?.isUp === true) {
+      isBridgeUp = true;
```
Comment (kevzzsk) [inline apps/web/src/pages/index.tsx:99]: i think you have to flip the logic here, if not `isBridgeUp` will never be `false`.

you need to check `if (data?.isUp === false)` then set `isBridgeUp` to false

**Review by kevzzsk — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/index.tsx (hunk 3 above)
Comment (nattadex) [reply apps/web/src/pages/index.tsx]: done!

File: apps/web/src/pages/index.tsx (hunk 2 above)
Comment (nattadex) [reply apps/web/src/pages/index.tsx]: donee!

File: apps/web/src/pages/index.tsx [hunk 4]
```diff
@@ -90,4 +90,29 @@ function Home() {
...
 
+export async function getServerSideProps() {
+  let isBridgeUp = true;
+
+  return fetch(`https://wallet.defichain.com/api/v0/bridge/status`)
+    .then((res) => Promise.all([res.json(), res.status]))
+    .then(([data, statusCode]) => {
+      if (statusCode === 200) {
+        if (data?.isUp === false) {
+          isBridgeUp = false;
```
Comment (thedoublejay) [inline apps/web/src/pages/index.tsx:101]:
```suggestion
          isBridgeUp = data?.isUp === false;
```

No need for the if statement anymore.

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**